### PR TITLE
cmd: implement explain command

### LINKED ISSUE #359: kubicorn explain
Using the reconciler pattern we can pull data about what is **Actually** running and what is **Expected** to be running...

Being able to run `kubicorn explain` and get a list of resources for each **actual** and **expected** would be handy!

We can output them as a table `-o table`
or as json `--output json` `-o json`
etc
etc

If you'd like to learn more about `kubicorn` project, you can read [The Project Walkthrough](http://kubicorn.io/documentation/readme.html).
If you have any question, comment on the issue, join us on the `#kubicorn` channel in the [Gophers Slack community](https://invite.slack.golangbridge.org/) or open a PR. We'll always do everything we can to get the PR though.

### PR DESCRIPTION
**Pull request by xmudrii**
Closes #359 

Currently, it only supports JSON format. Once we get this in I'll start working on the textual output as well.

Signed-off-by: Marko Mudrinić <mudrinic.mare@gmail.com>

### CODE REVIEW

File: cmd/explain.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+type OutputData struct {
+	Actual   *cluster.Cluster
+	Expected *cluster.Cluster
+}
+
+var exo = &ExplainOptions{}
+
+// ListCmd represents the list command
```
Comment (nrb) [inline cmd/explain.go]: Looks like a  doc/code mismatch  on this function name.

Comment (xmudrii) [reply cmd/explain.go]: @nrb Fixed and rebased PR

**Comment by kris-nova** [comment]
Can we update the conflicts? And thanks for updating the docs <3

**Comment by xmudrii** [comment]
rebased

**Pull request merged**